javaagent-core/src/main/java/org/hypertrace/agent/core/config/HypertraceConfig.java

```diff
@@ -51,7 +51,8 @@ private HypertraceConfig() {}
   // so avoiding for perf reasons
   private static volatile boolean servletCausingException;
 
-  private static AgentConfig agentConfig;
```

Comment:
Thanks for putting this together. I like the testability, but on the other hand, I would like to keep it as simple as possible. Would it be enough to just use `volatile` on this property and not use the supplier approach?

Reply:
Yeah, no problem! I think in addition to making this field `volatile`, we would also want to synchronize the write to this field in `HypertraceConfig.reset()` unless you're not worried about this class being thread-safe when APIs exposed for testing purpose are in use.